src/main/java/com/conveyal/datatools/manager/models/transform/DeleteRecordsTransformation.java

```diff
@@ -0,0 +1,72 @@
+package com.conveyal.datatools.manager.models.transform;
+
+import com.conveyal.datatools.common.status.MonitorableJob;
+import com.conveyal.datatools.manager.models.Snapshot;
+import com.conveyal.datatools.manager.persistence.Persistence;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Collections;
+import java.util.List;
+
+import static com.conveyal.datatools.manager.DataManager.GTFS_DATA_SOURCE;
+
+/**
+ * For now this is somewhat of just a demonstration/example for how to build create a {@link DbTransformation} subclass.
+ * This feed transformation will operate on the namespace for the provided {@link FeedTransformDbTarget#snapshotId}.
+ * It will delete all records in the specified table that match the WHERE clause created by the match field and values.
+ *
+ * TODO: Determine if this transformation class should be deleted. WARNING: this currently does not handle cascade
+ *   deleting stop_times associated with deleted trips (or records from other related tables).
+ */
+public class DeleteRecordsTransformation extends DbTransformation {
+    private static final Logger LOG = LoggerFactory.getLogger(DeleteRecordsTransformation.class);
+
+    public static DeleteRecordsTransformation create(String table, String matchField, List<String> matchValues) {
+        DeleteRecordsTransformation transformation = new DeleteRecordsTransformation();
+        transformation.table = table;
+        transformation.matchField = matchField;
+        transformation.matchValues = matchValues;
+        return transformation;
+    }
+
+    @Override
+    public void transform(FeedTransformTarget target, MonitorableJob.Status status) {
+        if (!(target instanceof FeedTransformDbTarget)) {
+            status.fail("Target must be FeedTransformDbTarget.");
+            return;
+        }
```

Comment:
I don't think the polymorphism will allow this because we need the `FeedTransformTarget` param in the transform method signature to match the super class

Reply:
Could this work?
```java
class FeedTransformation<T extends FeedTransformTarget> ...
    public abstract void transform(T target, ...)
```
followed by
```java
public abstract class DbTransformation extends FeedTransformation<FeedTransformDbTarget> {
...
public class DeleteRecordsTransformation extends DbTransformation {
    ...
    @Override
    public void transform(FeedTransformDbTarget target, ...) {

```